dyc/methods.py

```diff
@@ -139,11 +139,15 @@ def is_first_line_documented(self, result):
         str line: The line of the found method
         """
         returned = False
-        for x in range(result.start, result.end):
-            line = linecache.getline(result.filename, x)
-            if self.config.get("open") in line:
-                returned = True
-                break
+        read_first_line=linecache.getline(result.filename, result.start)
+        read_second_line=linecache.getline(result.filename, result.start+1)
+        finalTwoLines=read_first_line+"\n"+read_second_line
+        pattern = r'^[\s]*(def)\s[a-zA-Z0-9\_]*\([a-zA-Z0-9\_\,\s\=\[\]\(\)\{\}\*\&\%\!\-\"\'\+\;\.]*\)(\s\:|\:)[\n]*[\s]*(""")'  
```

Comment:
Can you help me understand this regex? Which pattern are we trying to catch?

Reply:
Earlier the issue was , if we have nested functions and the inner function is already documented, it was not asking user to enter the docstrings of the outer function even if it was not documented.
This problem was occuring because in earlier code we were just checking whether the function name stored in ( result ) variable has the presence of opening and closing triple quote braces """  . If triple quotes were present (even in nested inner function ), it simply returned True and user was not asked to document the outer method.

In my new solution code, we are trying to read the first first two lines of every function stored in (result) variable. and trying to match the pattern that :--    If any function starting with def keyword along with the declaration and after ():    has the triple quotes present immediately on the second line after some indentation , Then it should returned isdocumented as True() else False.

This pattern matching considers the Pep Style of docstrings, that the docstring should be first line after function /method definition .

If any other queries are there, Please let me know .

Reply:
@surbhiagrawal22 - These are the additional questions I have:

1. What if the line documented is not on the first line?
2. What if the documentation is in single quotes, most of these are defined in `config.yml` file. So perhaps we should get them from there?

```python
def hello():

    
    """
    Test Doc
    """
    print('World')

```

Reply:
1.  Since it take first and second line into consideration, the code takes into consideration the comments should be present on second line.If it is not present on second line , it will again ask user for docstrings.
2.  I can change the code to take single quotes into consideration..currently it works for multiline line triple quote docstrings.

Reply:
The values of comments are derived from https://github.com/Zarad1993/dyc/blob/master/dyc/configs/defaults.yaml . Maybe we can extract those values and make the pattern based of it?

Reply:
Hello, sorry for replying late . Can you help me how we can proceed with it further. 

https://github.com/Zarad1993/dyc/blob/master/dyc/configs/defaults.yaml . 
open: '"""'
close: '"""' are already defined there